prime-router/src/main/kotlin/azure/ReportFunction.kt

```diff
@@ -290,18 +281,19 @@ class ReportFunction : Logging {
             ?: request.queryParameters[PAYLOAD_NAME_PARAMETER]
     }
 
-    // TODO: Make this so that we check sender's configuration if there is no param type, this is blocked by
-    //  adding the 'processingType' attribute to a sender
-    private fun processingType(request: HttpRequestMessage<String?>): ProcessingType {
+    private fun processingType(request: HttpRequestMessage<String?>, sender: Sender): ProcessingType {
         // uppercase first char, so it matches enum when 'async' is passed in
-        val processingTypeString = request.queryParameters.getOrDefault(PROCESSING_TYPE_PARAMETER, "Sync")
-            .replaceFirstChar { it.uppercase() }
-        val processingType = try {
-            ProcessingType.valueOf(processingTypeString)
-        } catch (e: IllegalArgumentException) {
-            ProcessingType.Sync
+        val processingTypeString = request.queryParameters.get(PROCESSING_TYPE_PARAMETER)
```

Comment:
Probably should toLower and then firstChar toUppercase here to remove possibility of future incorrect capitalization messing with this. Or we should make the matching case-indeterminate

Reply:
Indeterminate is fine; but let's _use_ lower case everywhere, for this.

Reply:
That does mean changing it elsewhere, since we are currently matching to 'Async'.  I agree though, no reason to use caps at all. I think this may have been a 'lets make it work and then come back and modify' situation that got pushed under the rug.